### ISSUE TITLE
react/prop-types validation error

### ISSUE THREAD

**Issue by fengerzh** [yannickcr/eslint-plugin-react#1228]
Consider the following code:
```
ResourcesScreen.navigationOptions = {
  tabBarLabel: 'Resources',
  tabBarIcon: ({ tintColor }) => (
    <Image
      source={iconResources}
      style={[styles.icon, { tintColor }]}
    />
  ),
};
```
The third line caused an error: `'tintColor' is missing in props validation (react/prop-types)`. How to validate this kind of props?

**Comment by ljharb** [comment]
```jsx
function TabBarIcon({ tintColor }) {
  return (
    <Image
      source={iconResources}
      style={[styles.icon, { tintColor }]}
    />
  );
}
TabBarIcon.propTypes = {
  tintColor: PropTypes.string,
};

ResourcesScreen.navigationOptions = {
  tabBarLabel: 'Resources',
  tabBarIcon: TabBarIcon,
};

**Comment by fengerzh** [comment]
Thank you. Yes, it works.

**Comment by justinh00k** [comment]
This doesn't seem ideal.

**Comment by ljharb** [comment]
@justinh00k in what way? always using propTypes is very much ideal.

**Comment by justinh00k** [comment]
Using propTypes is great, writing a whole new and unnecessary function just to make propTypes pass is not.

**Comment by ljharb** [comment]
@justinh00k in this case, you're already writing the function - this is just storying it in an intermediate variable (which in general, is a good thing). You can also do this:
```jsx
ResourcesScreen.navigationOptions = {
  tabBarLabel: 'Resources',
  tabBarIcon: Object.assign(function TabBarIcon({ tintColor }) {
    return (
      <Image
        source={iconResources}
        style={[styles.icon, { tintColor }]}
      />
    );
  }, {
    propTypes: {
      tintColor: PropTypes.string,
    },
  }),
};
```

**Comment by justinh00k** [comment]
I would definitely prefer that, and writing it as an arrow function, since it doesn't require any additional code (or writing propTypes for an existing React function), and prevents you from having to redeclare state and navigation properties.

**Comment by ljharb** [comment]
Writing it as an arrow function (especially in the example I provided) prevents a reliable function name, which hurts debugging. Components should never be arrow functions.

There is no other difference between an arrow or a non-arrow function in React.

**Comment by justinh00k** [comment]
It is written as an arrow function in the React Navigation docs.
[https://reactnavigation.org/docs/tab-based-navigation/]

**Comment by ljharb** [comment]
Sure, but that doesn't mean it's a good idea - it's just trendy to do that.
